### DISCUSSION TITLE
Valetudo 2023.01.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1681] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2023.01.0</h2></p>
</div>

Five new supported robots, Four new features and improvements, Three new icons, Two new firmwares, One new year

## Viomi is dead, long live Viomi

The rising prices in Germany served as motivation to look into the Viomis again after having ended support for them a few releases ago.
Even though they're pretty mediocre robots, they're also relatively inexpensive and abundantly available both new and used.

Thanks to your donations, I was able to buy a few test devices to change that. Being able to directly verify correctness, reproduce issues and debug solutions yourself greatly helped with motivation. Furthermore, as they already had code for support in Valetudo, this wasn't actually all too much work.

I'd say that the implementation is now mostly on the quality level of the other supported robots. No more unparsable maps :)
The firmware itself is still a bit slow and weird, but you can very well live with that for the price of the unit.

Additionally, the rooting process has been reworked to be way more similar to other supported robots.

### Plenty of Viomi

![image](https://user-images.githubusercontent.com/974410/211048617-d347acbe-5843-4072-abc7-35f0d5fc7410.png)

Another big announcement is that Viomi is actually many now.

As you may or may not have known, Viomi is not a real vacuum robot manufacturer. Instead, they're just ordering "Viomi"-branded robots from the ODM 3irobotix. Specifically in our case it's the CRL-200S vacuum robot which has Valetudo support under its Viomi identity. Viomi is not the only "brand" that does that. In fact, there are quite a few that all feature the same hardware specs.

A question we had in the back of our minds for a while now was: "Can you just flash those other robots with a Viomi firmware?"

Turns out: Yes, you can.
And so, with no additional changes in Valetudo required, you can now free even more robots from the cloud.

The following models have been tested so far by me:

- Proscenic M6 Pro
- Cecotec Conga 3290
- Cecotec Conga 3790

Additionally, these models look promising and should be evaluated in the future:

- Viomi SE
- All other Cecotec Conga 3****
- Wyze Robot Vacuum
- Commodore CVR 200
- Kärcher RCV 3

If you have any of those and would like to help, please ping me.

## Roborock updates

While I was busy figuring out the Viomis, Dennis once again looked into new and existing Roborocks.

Also, once again thanks to your donations, I was able to buy a used S5 Max (and only got scammed once). Not only was this used for testing Valetudo but also to write step-by-step FEL rooting instructions for the Valetudo docs, which should make the whole process easier to understand.

### Firmware updates

The latest firmware versions for the S5 Max and S7 will soon be available in the Dustbuilder.
These now include a mapping pass functionality, which is also supported by the latest version of Valetudo.

**Note:** As we don't have a test device for the S7 (yet?), we weren't able to verify that it works properly.

### Q7 Max support

Due to - you guessed it - donations, we were able to buy this robot to see if it can be rooted.
And yes, we're pleased to announce that the FEL rooting method also works with the Q7 Max.

![image](https://user-images.githubusercontent.com/974410/211048915-ca9e8ea3-0412-49d3-92de-510a8371a6eb.png)

The Q7 Max and it's Auto-empty-dock sibling Q7 Max+ are now fully supported by Valetudo.
Expect to see it in the Dustbuilder soon.

## Valetudo features

Apart from robot support, there were also quite a few other changes.

### Live Map rework

This release introduces a Map Mode selector to the Live Map. 
If you want to do something, first select what you want to do and then simply interact with the map.

![image](https://user-images.githubusercontent.com/974410/211048136-5df818e4-038b-4ca9-9425-010d4e67545a.png)

The reasoning behind this change is that a few releases ago, it became possible to select segments by simply tapping on them anywhere.
As this conflicted with the Go-To marker feature, that feature was moved to press-and-hold. This wasn't because it was a particularly good idea to use a long press for that but because there was no other idea what to use.

With this change, the UX should be significantly better especially for newcomers, since you're now being shown directly which interactions with the map are possible and how you can do them.

This is also mostly just a return to the idea @Jomik proposed back in early 2021 when rewriting the frontend in react.
Thanks once again!

### New Icons

The previously confusing "1x" button in the live map view now features an icon that should be self-explanatory.

![image](https://user-images.githubusercontent.com/974410/211049457-a2373023-b6e1-4895-846c-0e6d9ebd3940.png)

Also, Icons in the menu have been updated for consistency and easier understanding.

![image](https://user-images.githubusercontent.com/974410/211049550-53cfc2c4-5316-43b0-a8a0-58d1fcd97fd2.png)

### Welcome Dialog

Speaking of newcomers, there's now a welcome dialog that tells a new user what their next steps should be.
It also conveniently features all the correct controls for their robot to make getting started even easier.

![image](https://user-images.githubusercontent.com/974410/211049683-7d3d8501-7878-4715-a43f-fc746e50da9d.png)

If you first want to play around with Valetudo without permanently dismissing the dialog, you can also temporarily hide it.
It will then reappear on the next page refresh.

### Valetudo Options

A new sub-menu was introduced with this version of Valetudo

![image](https://user-images.githubusercontent.com/974410/211049727-b139db7f-e007-45c4-945f-9fa4812dc576.png)

#### Config reset

It's now possible to reset the configuration via the UI. This can be useful if you'd like to gift a rooted robot to someone.
Unfortunately, we've _slightly_ missed Christmas with this update.

If you'd like to see that welcome dialog again, resetting the config will allow you to do that.

#### Update Provider Settings

The UI now also allows you to select the update channel.
The default "Release" channel should be used by most people, however if you like to live dangerously and want to support the development by testing unreleased versions for bugs, you can pick "Nightly".

The updater will then continue to work as it did before but pull its updates from the selected channel.

## Tray Companion

![image](https://user-images.githubusercontent.com/974410/211048367-3c3c22ec-29af-4cc6-b428-ec96c018a513.png)

The Valetudo Tray Companion app now works on both Windows and Linux thanks to @NKnusperer.

[https://github.com/Hypfer/valetudo-tray-companion](https://github.com/Hypfer/valetudo-tray-companion)

## The usual

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

### Breaking Changes

- **ui**: Restructure router and add controls for update provider and configuration reset [`8706988`](http://github.com/Hypfer/Valetudo/commit/8706988949ffe4b9d212ed4abe45b5c714e5ba0a)

### Features

- **vendor.dreame**: Add mop dock autorepair trigger quirk [`9932b03`](http://github.com/Hypfer/Valetudo/commit/9932b03616e3631f60e9ada736011da714f448e2)
- **vendor.roborock**: Add support for the Q7 Max [`bb6b66f`](http://github.com/Hypfer/Valetudo/commit/bb6b66f9b43b9b9a41f43317c17878cc09505051)
- **vendor.roborock**: Implement RoborockMappingPassCapability [`8374951`](http://github.com/Hypfer/Valetudo/commit/837495117b2881b680d28e7d158fb5200167e6f6)
- ContainerEntity upsert should return the previous attribute [`38ccba5`](http://github.com/Hypfer/Valetudo/commit/38ccba5f66ac9524f1684fac6ef2a4fcaca10703)
- **ui**: Add proper robot options icon [`046087c`](http://github.com/Hypfer/Valetudo/commit/046087c1e64516b0e0db0a197aee5805df49803c)
- **ui**: Allow temporarily hiding the welcome dialog [`f1c7238`](http://github.com/Hypfer/Valetudo/commit/f1c7238bc77b0b3138bc43131ec76aa5c8c22e91)
- Allow selection of the update provider [`68efeb8`](http://github.com/Hypfer/Valetudo/commit/68efeb86c1d932f611bff24704d6469f81dac0f3)
- Add action to reset the config to defaults [`e7959a4`](http://github.com/Hypfer/Valetudo/commit/e7959a4b2478ac0772191f8b5fef6da960cbcf8c)
- **ui**: Reintroduce live map mode toggle [`27554be`](http://github.com/Hypfer/Valetudo/commit/27554bed0244437d1fbc8601fc76b1b6458c9d8c)
- **oobe**: Don't show welcome dialog for existing installations [`0695457`](http://github.com/Hypfer/Valetudo/commit/0695457d9c52451f7b0c6ea621efcbe95e1448d3)
- **ui**: Change some icons [`759288e`](http://github.com/Hypfer/Valetudo/commit/759288e2ce6bbf9836bafcdb9a2d13faed7da75a)
- **ui**: Add iterationsIcon to live map controls [`0180c65`](http://github.com/Hypfer/Valetudo/commit/0180c65906fb2faa7e2916507fb0f24cbbbbd504)
- **oobe**: Add welcome dialog [`a43f30f`](http://github.com/Hypfer/Valetudo/commit/a43f30f6ac9ec65a15722cb958153b7d740d38bd)
- **vendor.roborock**: Add implementations for Roborock S7 auto empty dock [`39207e8`](http://github.com/Hypfer/Valetudo/commit/39207e8b76238d2483fd4592cd6b87db9e00c6b4)
- **vendor.dreame**: D10S Pro [`0939cbd`](http://github.com/Hypfer/Valetudo/commit/0939cbdcf28bb84581b93507c997c5a24ac9bbaa)
- **vendor.viomi**: Introduce ViomiOperationModeControlCapability [`d764438`](http://github.com/Hypfer/Valetudo/commit/d7644388e63d6cd740381f31c04362fdade7f6ad)
- **vendor.viomi**: Add y_mopping quirk [`5ac847f`](http://github.com/Hypfer/Valetudo/commit/5ac847f683419b7da7065f1e4dcc1d7bb0904529)
- **vendor.3ir**: Introduce common 3irobotix map parser [`8f5013a`](http://github.com/Hypfer/Valetudo/commit/8f5013a743a50e168f725372dc9ddc82ec4a332c)
- **vendor.viomi**: Track active segments [`56b0bf6`](http://github.com/Hypfer/Valetudo/commit/56b0bf66da6a0d3cf08e20760bcbfdb4102f7d74)
- **vendor.viomi**: Refactoring, fixes and tests [`7882db4`](http://github.com/Hypfer/Valetudo/commit/7882db41a1912aad028a4d48e60877abf6f6615a)
- **ui**: remove redundant first confirmation dialog for fetching updates [`4d01657`](http://github.com/Hypfer/Valetudo/commit/4d01657e9cf196bda0b44fb3b051853c9fecdf6a)

### Fixes

- Handle additional content in iw command output [`e5a2791`](http://github.com/Hypfer/Valetudo/commit/e5a2791bbb93f379dafd20b772ee77799e54d3f5)
- **vendor.viomi**: Repoll map on state change if it is an active state [`aac44b4`](http://github.com/Hypfer/Valetudo/commit/aac44b4a4bf7b38527782e2c12e5517aecb76c25)
- **vendor.roborock**: Ignore unknown_method message caused by miio_client downgrade on newer roborocks [`226a938`](http://github.com/Hypfer/Valetudo/commit/226a9389842fa9998dde1f52bbecd67d8adf13b0)
- **vendor.roborock**: Fix map not instantly repolling on state change for newer roborocks [`559076d`](http://github.com/Hypfer/Valetudo/commit/559076d30fae470f1e2b20eaeffadeb36d3e1bbd)
- **miio**: Use non-0 uid for provisioning to fix a38 0866 provisioning [`68e6d35`](http://github.com/Hypfer/Valetudo/commit/68e6d35ba75d2a223a8d22d7c179c7330b34635d)
- **ui**: Welcome dialog should work especially when embedded [`f97184d`](http://github.com/Hypfer/Valetudo/commit/f97184daad10c794359593c58cb0772153153a3b)
- **vendor.dreame**: Add warning message workaround for the half-broken W10 MappingPass [`2471c59`](http://github.com/Hypfer/Valetudo/commit/2471c59d16f927c40281034e0ec55e0ed09fe99d)
- **ui**: Fix segment actions for live map not handling the new live map mode selection properly [`56193f8`](http://github.com/Hypfer/Valetudo/commit/56193f8cb3b2b5dc3366dac5d10729c166effca4)
- **updater**: Delete any downloaded binary when reconfiguring updater [`2a74157`](http://github.com/Hypfer/Valetudo/commit/2a74157605423c8a335cf55d88c782ec45559990)
- **ui**: Clamp segment intersection calculation to map area [`dec5efb`](http://github.com/Hypfer/Valetudo/commit/dec5efb8b3945023a45606b45836c92852c58a89)
- **ui**: Fix dock controls on very small screens [`492cb42`](http://github.com/Hypfer/Valetudo/commit/492cb4210e5226173a1a07153d381be71853f1d0)
- Reduce map polling interval when dealing with huge maps to prevent valetudo from going OOM [`718115c`](http://github.com/Hypfer/Valetudo/commit/718115cabac8a95e6da02d91cfa1249d05be97ab)
- **ui**: Remove confusing state timestamp from NTPConnectivity [`692085e`](http://github.com/Hypfer/Valetudo/commit/692085e10bdf7d533a38c5f7e691fcfdac2e712d)
- **updater**: Better error message logging on http errors [`b6e7820`](http://github.com/Hypfer/Valetudo/commit/b6e78206a7f4a0542328a4fc25a6ffec1387c5a7)
- **vendor.dreame**: Remove broken operation mode control for W10 and reintroduce mop only quirk [`8b8d7a2`](http://github.com/Hypfer/Valetudo/commit/8b8d7a27b922f0e39fdfd0a52ba082f1ca11820e)
- **vendor.3ir**: Fix segment name block parsing [`99897a5`](http://github.com/Hypfer/Valetudo/commit/99897a5f8c6f1a8cbb1d2695ecd335f2e7db5d0d)
- **vendor.viomi**: Don't raise three events for a single error [`f143319`](http://github.com/Hypfer/Valetudo/commit/f14331978cd2fa5fbb23c53856102b598cc42ef5)
- **ui**: Allow valid slashes in mqtt topic prefix [`1f06b86`](http://github.com/Hypfer/Valetudo/commit/1f06b862bea577c8c22a23243c02d26b8f90c92a)
- **ui**: Fix occasional unexpected scrollbars [`bafdb4d`](http://github.com/Hypfer/Valetudo/commit/bafdb4d606d4f0d3d7d45147cba8570251460fe6)
- **vendor.3ir**: Parse temporary maps [`f43a7ed`](http://github.com/Hypfer/Valetudo/commit/f43a7ed68c17cd9edcecd98396380c67e6d9f9f8)
- **vendor.3ir**: Fix map parser going oom if uniqueMapId is all 0 [`b351686`](http://github.com/Hypfer/Valetudo/commit/b3516867c2530de1ac73a2e97cb768612670f721)
- Add .catch() handler to previously unhandled promises [`db5aeca`](http://github.com/Hypfer/Valetudo/commit/db5aeca526feb7899de43a40d511d3a9b1fdec74)
- **miio**: Protect against invalid Wi-Fi passwords [`4c873b7`](http://github.com/Hypfer/Valetudo/commit/4c873b71827c9bf8830fc30b7a6d6ee4d1aae2ca)
- **mqtt**: Make sure that we only refresh handles while we're connected [`67e66a0`](http://github.com/Hypfer/Valetudo/commit/67e66a0ee38007fbe9a2e9dabedecb525ac9f266)

### Refactoring

- **vendor.3ir**: Minor cleanup [`aad973e`](http://github.com/Hypfer/Valetudo/commit/aad973e3175ad5cc6caca93475045b557ba8a2c5)
- **vendor.3ir**: Minor cleanup [`a151080`](http://github.com/Hypfer/Valetudo/commit/a151080aa072f0d402c2e88830de8e564ff30fac)

### Chores

- **release**: 2023.01.0 [`3d6ef60`](http://github.com/Hypfer/Valetudo/commit/3d6ef60758077fc0623474721031c7493e39640d)
- **vendor.roborock**: Formatting [`2798fed`](http://github.com/Hypfer/Valetudo/commit/2798fed2fda2a702066a99a5633ed3e0fc22cc3c)
- **build**: Bump to NodeJS v18.12.1 [`25b7061`](http://github.com/Hypfer/Valetudo/commit/25b7061b25effe5e0d3119896e20b6a142d4c212)
- **vendor.viomi**: Remove unused map parser [`0c7d757`](http://github.com/Hypfer/Valetudo/commit/0c7d75710a865326e015239375b9a4d03ad0f564)
- minor cleanup [`79229f9`](http://github.com/Hypfer/Valetudo/commit/79229f99f690358488d7f97db1fbe8ee6f4711c5)
- Add hint regarding maps and reboots to the updater [`69b0ae7`](http://github.com/Hypfer/Valetudo/commit/69b0ae7aafb9c3621f7c10d0b3ffbcbbc637bd6b)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2023.01.0'>Valetudo 2023.01.0</a>.</em>

**Comment by lukyjay** [comment]
This change log is more fun than usual! Hope this means you're still finding fun in developing valetudo 🙂 thanks as always

**Comment by N1co93** [comment]
Thank you for the FEL root instruction for the Roborock S5 Max. With the newest instruction I was able to root my S5 Max without any problem 👍

**Comment by zombielinux** [comment]
It looks like I'll be able to get my hands on a Wyze Robot Vacuum (Model WVCR200S) in the next few days. 

Let me know how/when you want to help port valetudo to it. 

P.S. The S5 Max has been working beautifully for years now.

**Reply by zombielinux** [reply]
To all: The Wyze Robot Vacuum (WVCR200S) flashed successfully with only a minor change in procedure. 

I've documented it in this PR: https://github.com/Hypfer/valetudo-crl200s-root/pull/3

**Reply by Hypfer** [reply]
Nice! Could you also update the valetudo docs with an image similar to the ones already there?
e.g. https://valetudo.cloud/pages/general/supported-robots.html#conga_3790

**Comment by DSDV** [comment]
First and foremost thanks for the great work and such a great project! <3

Not meant to be rude or anything.
Just as a question because i consider buying one for valetudo use. 
Is the Q7 Max still planned? Since it is mentioned here it will become supported but i cant find it on https://dustbuilder.dontvacuum.me/

**Reply by Hypfer** [reply]
Weird. I remember it being visible on the dustbuilder index page but for some reason it seems to have vanished :raised_eyebrow: 

Anyway, it's back now. You can find it here: https://builder.dontvacuum.me/_a38.html

**Reply by DSDV** [reply]
thank you <3 
ill order one now :D

**Reply by DSDV** [reply]
now it arrived and is loading (: 
any recommendations or tips ? ( last time i flashed a S5 max but with the uart method)
